data/src/main/kotlin/com/vultisig/wallet/data/repositories/BalanceRepository.kt

```
@@ -223,101 +224,112 @@ internal class BalanceRepositoryImpl @Inject constructor(
-            ThorChain -> {
-                val listCosmosBalance = thorChainApi.getBalance(address)
-                val balance = listCosmosBalance
-                    .find {
-                        it.denom.equals(coin.ticker, ignoreCase = true) ||
-                                it.denom.equals(coin.contractAddress, ignoreCase = true)
-                    }
-                balance?.amount?.toBigInteger() ?: 0.toBigInteger()
-            }
+        val balance = try {
+            getBalance(coin, address)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)